dataflow/src/main/java/org/checkerframework/dataflow/expression/MethodReferenceTarget.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.dataflow.expression;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.qual.Pure;
+
+/**
+ * The right-hand side of a <a
+ * href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.13">Java Method
+ * Reference expression</a>
+ *
+ * <p>The right-hand side of a Java method Reference expression can have the following forms:
+ *
+ * <ul>
+ *   <li>{@literal [TypeArguments] Identifier}, which may be represented by a standard {@link
+ *       JavaExpression}
+ *   <li>{@literal [TypeArguments] "new"}, which is a constructor call and cannot be represented by
+ *       an arbitrary {@link JavaExpression}
+ * </ul>
+ */
+public class MethodReferenceTarget {
+
+  /** The type arguments for this method reference target. */
+  private final @Nullable JavaExpression typeArguments;
```

Comment:
I'm very surprised to see that `typeArguments` is a `JavaExpression`.  Is this just a placeholder?

Reply:
Coming back to this, I agree that it should not be a `JavaExpression`, since it can be things like `Array<String>` or `int[]`, which are decidedly _not_ expressions.

Would using a simple `TypeMirror` here suffice?

Reply:
For the semantics, a `TypeMirror` is likely to suffice.  For asking questions like whether the expression is mutable/assignable/contains method calls, the CF needs more information about what is actually in the source code.